pkg/sidecar/identity/member.go

```diff
@@ -173,32 +132,10 @@ func (m *Member) GetSeeds(ctx context.Context, coreClient v1.CoreV1Interface, ex
 	res := make([]string, 0, len(externalSeeds)+1)
 	res = append(res, externalSeeds...)
 
-	// Assume nodes share broadcast address type, and it's immutable.
-	switch m.NodesBroadcastAddressType {
-	case scyllav1.BroadcastAddressTypeServiceLoadBalancerIngress:
-		if len(svc.Status.LoadBalancer.Ingress) < 1 {
-			return nil, fmt.Errorf("service %q does not have ingress status, despite external address being set as broadcasted", naming.ObjRef(svc))
-		}
-		if len(svc.Status.LoadBalancer.Ingress[0].IP) != 0 {
-			res = append(res, svc.Status.LoadBalancer.Ingress[0].IP)
-		} else if len(svc.Status.LoadBalancer.Ingress[0].Hostname) != 0 {
-			res = append(res, svc.Status.LoadBalancer.Ingress[0].Hostname)
-		} else {
-			return nil, fmt.Errorf("service %q does not have external address, despite being set as broadcasted", naming.ObjRef(svc))
-		}
-	case scyllav1.BroadcastAddressTypeServiceClusterIP:
-		if svc.Spec.ClusterIP == corev1.ClusterIPNone {
-			return nil, fmt.Errorf("service %q does not have ClusterIP address, despite being set as broadcasted", naming.ObjRef(svc))
-		}
-		res = append(res, svc.Spec.ClusterIP)
-	case scyllav1.BroadcastAddressTypePodIP:
-		if len(pod.Status.PodIP) == 0 {
-			return nil, fmt.Errorf("pod %q does not have IP address, despite being set as broadcasted", naming.ObjRef(pod))
-		}
-		res = append(res, pod.Status.PodIP)
-	default:
-		return nil, fmt.Errorf("unsupported node broadcast address type %q", m.NodesBroadcastAddressType)
+	localSeed, err := controllerhelpers.GetScyllaBroadcastAddress(m.NodesBroadcastAddressType, svc, pod)
```

Comment:
this seems to be called in the "constructor", maybe you can save the result on the object rather then the type?

Reply:
> maybe you can save the result on the object rather then the type?

could you please rephrase that? I've read it like 10 times and don't understand what you mean by that

Reply:
This function is already executed in `NewMember`, only the type is stored and the function is executed again here with `m.NodesBroadcastAddressType`. I am suggesting to cache the result instead of the argument (`m.NodesBroadcastAddressType`) and avoid running it a second time.
https://github.com/scylladb/scylla-operator/blob/53078ab596ba79e5f0b01475b31840f9aff6c5ce/pkg/sidecar/identity/member.go#L59-L65

Reply:
oh, it's called for a different node here (`pod := otherPods[0]`)

Reply:
thx, my bad. from the diff, it seemed like the same one

Reply:
I see that I accidentally moved the comment explaining the use of member's node broadcast address type for it (`// Assume nodes share broadcast address type and it's immutable.`) so I'll fix that